cmd/angryjet/main.go

```diff
@@ -133,6 +136,36 @@ func GenerateManaged(filename, header string, p *packages.Package) error {
 	return errors.Wrap(err, "cannot write managed resource methods")
 }
 
+// GenerateManagedV2 generates the resource.Managed method set for v2-style namespaced MRs.
+func GenerateManagedV2(filename, header string, p *packages.Package) error {
```

Comment:
WDYT about using e.g. `GenerateManagedLegacy` and `GenerateManagedModern` naming conventions here? That'd be more in line with what we did for XRs, for example:

https://github.com/crossplane/crossplane-runtime/blob/f5f608c93ded2ef2ea30daa38a6fd7aaed53efb2/pkg/resource/unstructured/composite/composite.go#L33-L47

I'd prefer to be consistent, but it's not a blocker.